test/DynamoCoreTests/Graph/Workspaces/WorkspaceModelTests.cs

```diff
@@ -0,0 +1,208 @@
+
+        [Test]
+        [Category("UnitTests")]
+        public void LoadLegacyNotes_AllNewNotesTest()
+        {
+            //Generating tests data
+            var mockViewBlock = new ExtraWorkspaceViewInfo();
+            mockViewBlock.Notes = new[] {
+                new ExtraNoteViewInfo()
+                { Id = Guid.NewGuid().ToString(),
+                    Text = "Test data text",
+                    X = 1,
+                    Y = 1
+                },
+                new ExtraNoteViewInfo()
+                { Id = Guid.NewGuid().ToString(),
+                    Text = "Test data text",
+                    X = 1,
+                    Y = 1
+                }
+            };
+
+            //Checks no notes exited before
+            Assert.AreEqual(0, this.CurrentDynamoModel.CurrentWorkspace.Notes.Count());
+
+            //Adds and checks if notes where added
+            this.CurrentDynamoModel.CurrentWorkspace.UpdateWithExtraWorkspaceViewInfo(mockViewBlock);
+            Assert.AreEqual(2, this.CurrentDynamoModel.CurrentWorkspace.Notes.Count());
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void LoadLegacyNotes_DuplicatedNoteIdTest()
+        {
+            Guid toBeDuplicatedId = Guid.NewGuid();
+
+            //Generating tests data
+            var mockViewBlock = new ExtraWorkspaceViewInfo();
+            var firstNote = new ExtraNoteViewInfo()
+            {
+                Id = toBeDuplicatedId.ToString(),
+                Text = "First test note data",
+                X = 1,
+                Y = 1
+            };
+            mockViewBlock.Notes = new[] { firstNote };
+
+            //Checks no notes exited before
+            Assert.AreEqual(0, this.CurrentDynamoModel.CurrentWorkspace.Notes.Count());
+
+            //Adds and checks that the note was added
+            this.CurrentDynamoModel.CurrentWorkspace.UpdateWithExtraWorkspaceViewInfo(mockViewBlock);
+            Assert.AreEqual(1, this.CurrentDynamoModel.CurrentWorkspace.Notes.Count());
+
+            //Loads a second note with the same id
+            mockViewBlock.Notes = new[] {
+                new ExtraNoteViewInfo()
+                { Id = toBeDuplicatedId.ToString(),
+                    Text = "Second test note data",
+                    X = 1,
+                    Y = 1
+                }
+            };
+            
+            this.CurrentDynamoModel.CurrentWorkspace.UpdateWithExtraWorkspaceViewInfo(mockViewBlock);
+            
+            //Checks that there is still only one note
+            Assert.AreEqual(1, this.CurrentDynamoModel.CurrentWorkspace.Notes.Count());
+
+            //Verifies that the remaining note is in fact the first one added
+            Assert.AreEqual(firstNote.Text, this.CurrentDynamoModel.CurrentWorkspace.Notes.FirstOrDefault().Text);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void CheckIfModelExistsInSameGroup_True_ShouldNotCreateNewAnnotation()
```

Comment:
@Astul-Betizagasti if you updated the function name, maybe update the test name here as well? Can you also put a comment using `///` to describe what function this if for?

Reply:
sure, on it